sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/session/HiveSessionImplwithUGI.java

```diff
@@ -85,15 +73,6 @@ public String getDelegationToken() {
     return this.delegationTokenStr;
   }
 
-  @Override
-  protected synchronized void acquire(boolean userAccess) {
-    super.acquire(userAccess);
-    // if we have a metastore connection with impersonation, then set it first
-    if (sessionHive != null) {
-      Hive.set(sessionHive);
```

Comment:
does it have side effects?

Reply:
Removing or existing？

Reply:
`Hive.set(sessionHive)` this code looks like having side effects. Is it really safe to remove it?

Reply:
In the Hive repo, it implements this logic in its superclass `HiveSessionImpl`. In the Spark repo, we have this in HiveSessionImplwithUGI only but we mostly use `HiveSessionImpl` w/o this.

The Hive metadata operations rely upon this, but on the spark side, the metadata operations have been overridden with catalog APIs.